common/src/chain/transaction/signature/tests/mod.rs

```diff
@@ -9,308 +15,229 @@ use crate::{
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::SignatureNotFound)
-        );
-    }
-    {
-        // SigHashType ALL - must fail because there are no bytes in raw_signature
-        tx.update_witness(
-            0,
-            InputWitness::Standard(StandardInputSignature::new(
-                SigHashType::try_from(SigHashType::ALL).unwrap(),
-                vec![],
-            )),
-        )
-        .unwrap();
-        assert_eq!(
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::InvalidSignatureEncoding)
-        );
-    }
-    {
-        // SigHashType ALL - must fail because there are wrong bytes in raw_signature
-        tx.update_witness(
-            0,
-            InputWitness::Standard(StandardInputSignature::new(
-                SigHashType::try_from(SigHashType::ALL | SigHashType::ANYONECANPAY).unwrap(),
-                vec![1, 2, 3, 4, 5, 6, 7, 8, 9],
-            )),
-        )
-        .unwrap();
-        assert_eq!(
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::InvalidSignatureEncoding)
-        );
-    }
-    {
-        // SigHashType NONE - must fail because there are no bytes in raw_signature
-        tx.update_witness(
-            0,
-            InputWitness::Standard(StandardInputSignature::new(
-                SigHashType::try_from(SigHashType::NONE).unwrap(),
-                vec![],
-            )),
-        )
-        .unwrap();
-        assert_eq!(
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::InvalidSignatureEncoding)
-        );
-    }
-    {
-        // SigHashType NONE - must fail because there are wrong bytes in raw_signature
-        tx.update_witness(
-            0,
-            InputWitness::Standard(StandardInputSignature::new(
-                SigHashType::try_from(SigHashType::NONE | SigHashType::ANYONECANPAY).unwrap(),
-                vec![1, 2, 3, 4, 5, 6, 7, 8, 9],
-            )),
-        )
-        .unwrap();
-        assert_eq!(
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::InvalidSignatureEncoding)
-        );
-    }
-    {
-        // SigHashType SINGLE - must fail because there are no bytes in raw_signature
-        tx.update_witness(
-            0,
-            InputWitness::Standard(StandardInputSignature::new(
-                SigHashType::try_from(SigHashType::SINGLE).unwrap(),
-                vec![],
-            )),
-        )
-        .unwrap();
-        assert_eq!(
-            verify_signature(&outpoint_dest, &tx, 0),
-            Err(TransactionSigError::InvalidSignatureEncoding)
+            verify_signature(&destination, &tx, 0),
+            Err(TransactionSigError::SignatureNotFound),
+            "{destination:?}"
         );
     }
+}
+
+// Try to verify empty and invalid signatures.
+#[test]
+fn verify_invalid_signature() {
```

Comment:
I feel the word "invalid" is overused. Yes, the signatures are invalid. But it's preferable to say what it's that's invalid in the signature. Is it empty? Mutated? Invalid encoding? and so on.